Objects/StrategyCardAutomator.ttslua

```diff
@@ -318,68 +355,104 @@ local backManual = {
 	}
 }
 
+local backManual = {
+	tag = "Panel",
+	attributes = {
+		position = '0 0 -20.5',
+		rotation = '0 0 180',
+		width = 200,
+		height = 270,
+		scale = '1.35 0.709 1.0',
+	},
+	children = {
+		{
+			tag = "Text",
+			attributes = {
+				resizeTextForBestFit = "true",
+				alignment = "UpperLeft",
+				horizontalOverflow = "wrap"
+			},
+			value = _ObjectManual
+		}
+	}
+}
+
 local _updateUiText = "Please click the 'Update UI' button after all players have chosen a faction. Adds automation features (on first use) and updates the colors listed under politics and trade."
 
-local objectUI = {{
+local objectUI = {
 	tag = "VerticalLayout",
 	attributes = {
-		position = '0 0 2',
-		rotation = '0 180 0',
+		position = '20 24 2',
+		rotation = '0 180 180',
 		width = 200,
-		height = 280,
+		height = 20,
 		padding = '10 10 10 10',
 		spacing = 10,
 		scale = '1.35 1.0 0.709'
 	},
 	children = {
 		{
-			tag ="Text",
+			tag = 'Toggle',
 			attributes = {
-				fontSize = 25,
-				textStyle = "Bold",
-				alignment = "MiddleCenter"
+				id = 'AutomateSecretsToggle',
+				onValueChanged = 'onToggleValueChanged',
+				default = 'false',
+				fontSize = 14,
 			},
-			value = "Strategy Card Automator"
-		},
+			value = "Don't deal Secrets"
+		}
+	},
+}
+
+local backUI = {
+	tag = "Panel",
+	attributes = {
+		position = '0 0 -20.5',
+		rotation = '0 0 180',
+		width = 200,
+		height = 270,
+		scale = '1.35 0.709 1.0',
+	},
+	children = {
 		{
-			tag = "Button",
+			tag = "Text",
 			attributes = {
-				onClick = "updateUiOnClick",
-				fontSize = 18
+				resizeTextForBestFit = "true",
+				alignment = "UpperLeft",
+				horizontalOverflow = "wrap"
 			},
-			value = "Update UI"
-		},
-		{
-			tag = "Panel",
-			children = {
-				{
-					tag = "Text",
-					attributes = {
-						resizeTextForBestFit = "true",
-						alignment = "UpperLeft",
-						horizontalOverflow = "wrap"
-					},
-					value = _updateUiText
-				}
-			}
+			value = _ObjectManual
 		}
 	}
-}}
+}
+
+function onToggleValueChanged(player, value, id)
+	local valueAsBool = string.lower(value) == 'true' and true or false
+	_dontDealSecrets = valueAsBool
+	if _dontDealSecrets then
+		broadcastToAll("No longer automatically dealing Secrets for Imperial Secondary.", 'Red')
+	else
+		broadcastToAll("Automatically dealing Secrets for Imperial Secondary.", 'Green')
+	end
+end
 
 function onLoad()
-	--Description on Back of the Object.
-	self.UI.setXmlTable({backManual})
+	_dontDealSecrets = false
```

Comment:
This works, but it's good practice to define this as a global variable near the top and set to false there.  No need to for onLoad to reset values for simple variables.

Reply:
Also it's good practice to use the positive form, so "_dealSecrets = true" instead of "don't"